packages/core/stories/multiline-input/multiline-input.stories.tsx

```diff
@@ -104,26 +107,30 @@ export const Disabled: ComponentStory<typeof MultilineInput> = (args) => {
 
 export const CharacterCount: ComponentStory<typeof MultilineInput> = (args) => {
   const [value, setValue] = useState<string>("Value");
+  const [isError, setIsError] = useState<boolean>(false);
   const MAX_CHARS = 10;
 
   const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
     const newVal = event.target.value;
+    setValue(newVal);
 
-    if (newVal.length <= MAX_CHARS) {
-      const value = newVal;
-      setValue(value);
+    if (newVal.length > MAX_CHARS) {
+      setIsError(true);
+    } else if (isError) {
+      setIsError(false);
     }
   };
 
   return (
     <MultilineInput
       {...args}
-      value={value}
-      onChange={handleChange}
+      className={clsx({ error: isError })}
       endAdornment={
         <Label variant="secondary">{`${value.length}/${MAX_CHARS}`}</Label>
```

Comment:
Is there a preference on providing custom component v.s. using an out of box different Text component when error? If it's not achievable from Text component, is there a need to review the design?

Reply:
changed implementation